crates/nu-command/src/core_commands/def_env.rs

```diff
@@ -28,7 +28,33 @@ impl Command for DefEnv {
 
     fn extra_usage(&self) -> &str {
         r#"This command is a parser keyword. For details, check:
-  https://www.nushell.sh/book/thinking_in_nushell.html"#
+  https://www.nushell.sh/book/thinking_in_nushell.html
+
+=== EXTRA NOTE ===
+All blocks are scoped, includes variable definition and environment variable changes.
+
+Because of this, the following doesn't works for you:
```

Comment:
```suggestion
All blocks are scoped, including variable definition and environment variable changes.

Because of this, the following doesn't work:
```